### PR TITLE
Remove COPY (query) limitation for ON SEGMENT clause

### PR DESCRIPTION
**Pull request by dyozie**
This change will also go into 6X_STABLE, and into 5X_STABLE after the code backport is complete.

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 1]
```diff
@@ -367,83 +367,85 @@ IGNORE EXTERNAL PARTITIONS</codeblock></p>
...
+        in binary format.</p>
+      <p>You must have <codeph>SELECT</codeph> privilege on the table whose values are read by
+          <codeph>COPY TO</codeph>, and insert privilege on the table into which values are inserted
+        by <codeph>COPY FROM</codeph>. It is sufficient to have column privileges on the columns
+        listed in the command.</p>
+      <p>Files named in a <codeph>COPY</codeph> command are read or written directly by the database
+        server, not by the client application. Therefore, they must reside on or be accessible to
+        the Greenplum Database master host machine, not the client. They must be accessible to and
+        readable or writable by the Greenplum Database system user (the user ID the server runs as),
+        not the client. <codeph>COPY</codeph> naming a file is only allowed to database superusers,
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: *Only database superusers are permitted to COPY to or from a file,* ?

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 2]
```diff
@@ -367,83 +367,85 @@ IGNORE EXTERNAL PARTITIONS</codeblock></p>
...
+        time the files were generated with the <codeph>COPY TO</codeph> command. The
+          <codeph>COPY</codeph> command might return table distribution policy errors, if you
+        attempt to restore table data and the table distribution policy was changed after the
+          <codeph>COPY FROM...ON SEGMENT</codeph> was run.</p>
+      <note>If you run <codeph>COPY FROM...ON SEGMENT</codeph> and the server configuration
+        parameter <codeph>gp_enable_segment_copy_checking</codeph> is <codeph>false</codeph>, manual
+        redistribution of table data might be required. See the <codeph>ALTER TABLE</codeph> clause
+          <codeph>WITH REORGANIZE</codeph>.</note>
+      <p>When you specify the <codeph>LOG ERRORS</codeph> clause, Greenplum Database captures errors
+        that occur while reading the external table data. You can view and manage the captured error
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: there are a few references to *external table*.  one instance says COPY can't be used with external tables.  the next reference discusses how to deal with child partitions that are external tables.  is this reference to external table correct?

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 3]
```diff
@@ -470,14 +472,16 @@ IGNORE EXTERNAL PARTITIONS</codeblock></p>
...
-        command, the command can be controlled by a resource queue. The resource queue must be
-        configured with the <codeph>ACTIVE_STATEMENTS</codeph> parameter that specifies a maximum
-        limit on the number of queries that can be executed by roles assigned to that queue.
-        Greenplum Database does not apply a cost value or memory value to a <codeph>COPY</codeph>
-        command, resource queues with only cost or memory limits do not affect the running of
-          <codeph>COPY</codeph> commands.</p><p>A non-superuser can run only these types of
-          <codeph>COPY</codeph> commands:<ul id="ul_bf3_2pk_zs">
+      </ul>
+      <p>When a Greenplum Database user who is not a superuser runs a <codeph>COPY</codeph> command,
+        the command can be controlled by a resource queue. The resource queue must be configured
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: wondering why there is a specific reference to resource queues here, but no reference to resource groups?

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml (hunk 2 above)
Comment (dyozie) [reply gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: I think only gpcopy has that restriction spelled out.  At any rate, COPY has enough references to external tables here that I don't want to start second guessing it.

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml (hunk 3 above)
Comment (dyozie) [reply gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: It is strange, but I'm hesitant to add RGs here unless we know for sure they would apply.

**Comment by dyozie** [comment]
https://github.com/greenplum-db/gpdb/pull/7726 was merged into 5X_STABLE, so merging this related doc change.

**Pull request merged**